functions/src/events/scrapeEvents.ts

```diff
@@ -138,12 +146,60 @@ class HearingScraper extends EventScraper<HearingListItem, Hearing> {
     return events.filter(HearingListItem.guard)
   }
 
-  async getEvent({ EventId }: HearingListItem) {
-    const content = HearingContent.check(await api.getHearing(EventId))
+  async getEvent({ EventId }: HearingListItem /* e.g. 4962 */) {
+    const data = await api.getHearing(EventId)
+    const content = HearingContent.check(data)
+    const eventInDb = await db.collection("events").doc(String(EventId)).get()
+    const hearing = Hearing.check(eventInDb)
+    const newToken = randomBytes(16).toString("hex")
+
+    let maybeVideoURL = null
+    let transcript = null
+    if (!hearing.videoFetchedAt) {
+      const req = await fetch(
+        `https://malegislature.gov/Events/Hearings/Detail/${EventId}`
+      )
+      const res = await req.text()
+      if (res) {
+        const dom = new JSDOM(res)
+        if (dom) {
+          const maybeVideoSource =
+            dom.window.document.querySelectorAll("video source")
```

Comment:
It's just occurred to me that with the way this is structured, we're scraping + hitting AssemblyAI *before* the cutoff check, so this is in theory running for every hearing in history. We should probably refactor this so that the order of events is:
1. Fetch the Hearing from the MA Legislature API
2. Check the startTime cutoff and quit early if we fail the check
3. Only now that we're sure that it's a relevant event, actually scrape the Hearing page for the video and make our AssemblyAI call

Reply:
Updated to only transcribe hearings that are 8 days old or less.